### PR TITLE
tintin++: add new package

### PR DESCRIPTION
**Pull request by wzyou**
Signed-off-by: wzyou <wzyou.goo@gmail.com>

Maintainer: me / @\<github-user> (find it by checking history of the package Makefile)
Compile tested: (put here arch, model, OpenWrt version)
Run tested: (put here arch, model, OpenWrt version, tests done)

Description:

### CODE REVIEW

**Comment by PolynomialDivision** [comment]
We have a "real name" policy. That is why the CI fails.

File: utils/tintin/Makefile [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+PKG_NAME:=tintin
+PKG_VERSION:=2.02.20
+PKG_RELEASE:=$(AUTORELEASE)
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://github.com/scandum/tintin/releases/download/$(PKG_VERSION)/
+PKG_HASH:=331673e6ee3c945cf27e1c0d71cec1225c9d992588ed73b2a707c4c49523e8d2
+
+PKG_MAINTAINER:=scandum <ivdhoven@gmail.com>
```
Comment (BKPepe) [inline utils/tintin/Makefile:11]: Who is this? That's not you. You copy&pasted it from somewhere, right? Does he agreed that he is going to be maintainer of this package in this repository?

File: utils/tintin/Makefile [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/tintin
+  SECTION:=utils
+  CATEGORY:=Utilities
+  SUBMENU:=Terminal
+  TITLE:=tintin a MUD client
+  DEPENDS:=+zlib +libpcre +libgnutls
+  URL:=https://downloads.sf.net/tintin/
```
Comment (BKPepe) [inline utils/tintin/Makefile:27]: Isn't homepage this https://tintin.mudhalla.net/  ?

File: utils/tintin/Makefile [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  PCRE, UTF-8, SSL, VT100, IPv6, the dynamic TINTIN scripting language,
+  a comprensive trigger system, and several other features geared towards
+  mudding, scripting, shell-scripting, text processing, automation,
+  interprocess communication, redirection, xterm terminal emulation, and
+  multiplexing / demultiplexing.
+endef
+
+define Build/Prepare
+	$(Build/Prepare/Default)
+	($(CP) $(PKG_BUILD_DIR)/../tt/src/. $(PKG_BUILD_DIR)/ && \
```
Comment (BKPepe) [inline utils/tintin/Makefile:41]: I wonder what are you trying to achieve with this. Do you have any error logs?

**Review by BKPepe — CHANGES_REQUESTED** [review]
There are just more changes required than "real name" policy as said. 

1. PR template was not filled.
- Was it compile and run tested?
2. I don't know why this package is required and should be added to this repository.
- That's can be answered in the commit description or what is your use case exactly since I don't like that new packages are added, then abandoned and then removed.

You already added it to downstream distribution (https://github.com/coolsnowwolf/packages/pull/418) .

File: utils/tintin/Makefile (hunk 1 above)
Comment (wzyou) [reply utils/tintin/Makefile:11]: I'm going to be maintainer of this package in this repository. I haven't figured out what PKG_MAINTAINER means before

File: utils/tintin/Makefile (hunk 2 above)
Comment (wzyou) [reply utils/tintin/Makefile:27]: yes

File: utils/tintin/Makefile (hunk 3 above)
Comment (wzyou) [reply utils/tintin/Makefile:41]: in the default prepare script, compile in directory PKG_BUILD_DIR after unzip . But this .tar.gz will unzip to directory PKG_BUILD_DIR/../tt/

**Comment by wzyou** [comment]
this is a MUD terminal client, to play MUD games. I need it but if it shouldn't be added to this  repository, I will close this pull request.

File: utils/tintin/Makefile (hunk 3 above)
Comment (wzyou) [reply utils/tintin/Makefile:41]: error log:
```shell
make[3]: Entering directory '/home/user/openwrt/openwrt/build_dir/target-mipsel_24kc_musl/tintin-2.02.20'
make[3]: *** No targets specified and no makefile found.  Stop.
make[3]: Leaving directory '/home/user/openwrt/openwrt/build_dir/target-mipsel_24kc_musl/tintin-2.02.20'
make[2]: *** [Makefile:53: /home/user/openwrt/openwrt/build_dir/target-mipsel_24kc_musl/tintin-2.02.20/.built] Error 2
make[2]: Leaving directory '/home/user/openwrt/openwrt/feeds/packages/utils/tintin'
time: package/feeds/packages/tintin/compile#0.55#0.03#0.57
    ERROR: package/feeds/packages/tintin failed to build.
make[1]: *** [package/Makefile:116: package/feeds/packages/tintin/compile] Error 1
make[1]: Leaving directory '/home/user/openwrt/openwrt'
make: *** [/home/user/openwrt/openwrt/include/toplevel.mk:230: package/tintin/compile] Error 2
# user @ P3TERX in ~/openwrt/openwrt on git:master x [7:56:11] C:2
$ ls /home/user/openwrt/openwrt/build_dir/target-mipsel_24kc_musl/tintin-2.02.20

# user @ P3TERX in ~/openwrt/openwrt on git:master x [7:57:22] 
$ ls /home/user/openwrt/openwrt/build_dir/target-mipsel_24kc_musl/tt            
COPYING  CREDITS  FAQ  INSTALL  NEWS  README  SCRIPTS  TODO  docs  mods  src

# user @ P3TERX in ~/openwrt/openwrt on git:master x [7:57:28] 
$ 
```
and configure error:
```shell
checking for gnutls_init in -lgnutls... yes
checking for sqrt in -lm... yes
checking for gmake... no
checking for make... make
checking for /dev/ptmx... configure: error: cannot check for file existence when cross compiling
make[2]: *** [Makefile:54: /home/user/openwrt/openwrt/build_dir/target-mipsel_24kc_musl/tintin-2.02.20/.configured_68b329da9893e34099c7d8ad5cb9c940] Error 1
make[2]: Leaving directory '/home/user/openwrt/openwrt/feeds/packages/utils/tintin'
time: package/feeds/packages/tintin/compile#2.05#0.54#2.35
    ERROR: package/feeds/packages/tintin failed to build.
make[1]: *** [package/Makefile:116: package/feeds/packages/tintin/compile] Error 1
make[1]: Leaving directory '/home/user/openwrt/openwrt'
make: *** [/home/user/openwrt/openwrt/include/toplevel.mk:230: package/tintin/compile] Error 2
```

Comment (jefferyto) [reply utils/tintin/Makefile:41]: If you add:

```make
TAR_OPTIONS+= --strip-components 1
TAR_CMD=$(HOST_TAR) -C $(1) $(TAR_OPTIONS)
```

to the Makefile then it should extract into the correct directory.

**Comment by BKPepe** [comment]
Closing for now. If you are willing to do suggested changes, feel free to reopen this.

**Pull request closed without merging**